aiocache/base.py

```diff
@@ -109,18 +110,26 @@ class BaseCache:
     NAME: str
 
     def __init__(
-        self, serializer=None, plugins=None, namespace=None, key_builder=None, timeout=5, ttl=None
+        self,
+        serializer: Optional[BaseSerializer] = None,
+        plugins: Optional[List[Any]] = None,  # aiocache.plugins depends on aiocache.base
+        namespace: str = "",
+        key_builder: Optional[Callable[[str, str], str]] = None,
+        timeout: Optional[float] = 5,
+        ttl: Optional[float] = None,
     ):
-        self.timeout = float(timeout) if timeout is not None else timeout
-        self.namespace = namespace
-        self.ttl = float(ttl) if ttl is not None else ttl
-        self.build_key = key_builder or self._build_key
+        self.timeout: Optional[float] = float(timeout) if timeout is not None else None
+        self.ttl: Optional[float] = float(ttl) if ttl is not None else None
 
-        self._serializer = None
-        self.serializer = serializer or serializers.StringSerializer()
+        self.namespace: str = namespace
+        self._build_key: Callable[[str, str], str] = key_builder or (
```

Comment:
```suggestion
        self._build_key = key_builder or (
```
Hopefully this one too.

Reply:
Mypy is fine without this annotation, but it might be useful for documenting the code?  This is because lambdas can't have type annotations.

Reply:
> Frankly, you could just define this lambda as the parameter default and disallow `None` completely.

I like this suggestion, but mypy does not. :(

Reply:
type: ignore it if it's a bug in mypy.

I know there's this issue around lambdas already: https://github.com/python/mypy/issues/9590

Reply:
It turns out this was my error.  I think I must have missed a comma at the end of the line or something mundane like that.  Correcting it now.  Will push in the next round of updates for this PR.